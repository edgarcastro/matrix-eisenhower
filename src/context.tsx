import React, { Dispatch, SetStateAction, useState } from 'react';

interface ContextModel {
  email?: string | null;
  displayName?: string | null;
  setEmail: Dispatch<SetStateAction<string | undefined | null>>;
  setDisplayName: Dispatch<SetStateAction<string | undefined | null>>;
}

const defaultValue: ContextModel = {
  email: '',
  displayName: '',
  setEmail() {
    //Do nothing
  },
  setDisplayName() {
    //Do nothing
  },
};

export const AppContext: React.Context<ContextModel> =
  React.createContext(defaultValue);

type AppProviderProps = {
  children: React.ReactNode;
};

export const AppProvider: React.FC<AppProviderProps> = ({ children }) => {
  const [email, setEmail] = useState<string | undefined | null>('');
  const [displayName, setDisplayName] = useState<string | undefined | null>('');

  const value: ContextModel = {
    email,
    displayName,
    setEmail,
    setDisplayName,
  };

  return <AppContext.Provider value={value}>{children}</AppContext.Provider>;
};
