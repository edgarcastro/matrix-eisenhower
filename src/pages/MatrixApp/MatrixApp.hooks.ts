import createStyles from '@mui/styles/createStyles';
import makeStyles from '@mui/styles/makeStyles';

export const useStyles = makeStyles(theme =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
  })
);
