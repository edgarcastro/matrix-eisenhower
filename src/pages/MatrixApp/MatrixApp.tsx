import React, { useContext } from 'react';
import { AppBar, Toolbar, Typography, Button } from '@material-ui/core';
import Matrix from '../../components/Matrix';
import { signOut } from '../../api';
import { APP_NAME } from '../../constants';
import { useStyles } from './MatrixApp.hooks';
import { AppContext } from '../../context';

const MatrixApp: React.FC = () => {
  const classes = useStyles();
  const { displayName } = useContext(AppContext);

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            {APP_NAME}
          </Typography>
          {displayName ? `Hola, ${displayName}` : ''}
          <Button onClick={signOut} color="inherit">
            Sign Out
          </Button>
        </Toolbar>
      </AppBar>
      <Matrix />
    </div>
  );
};

export default MatrixApp;
