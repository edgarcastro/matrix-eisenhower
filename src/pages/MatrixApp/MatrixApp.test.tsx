import React from 'react';
import MatrixApp from './MatrixApp';
import { shallow } from 'enzyme';
import { AppProvider } from '../../context';

function renderMatrixApp() {
  return shallow(
    <AppProvider>
      <MatrixApp />
    </AppProvider>
  );
}

describe('<MatrixApp>', () => {
  test('Expect the component match with the snapshot', () => {
    const wrapper = renderMatrixApp();
    expect(wrapper).toMatchSnapshot();
  });
});
