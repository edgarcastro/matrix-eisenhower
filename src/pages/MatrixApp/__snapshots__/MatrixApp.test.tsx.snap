// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatrixApp> Expect the component match with the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "displayName": "",
      "email": "",
      "setDisplayName": [Function],
      "setEmail": [Function],
    }
  }
>
  <MatrixApp />
</ContextProvider>
`;
